Downsample/Downsample.cpp

```diff
@@ -138,17 +138,26 @@ Eigen::MatrixXf gridSum(const Eigen::MatrixXf &r, int downsamplingFactor)
 
 Eigen::MatrixXf lineSum(const Eigen::MatrixXf &matrix, int downsamplingFactor)
 {
+    Eigen::Map<Eigen::MatrixXf> reshapedMatrixMap(const_cast<float *>(matrix.data()),
+                                                  downsamplingFactor,
+                                                  (int)matrix.rows() * (int)matrix.cols() / downsamplingFactor);
     Eigen::MatrixXf reshapedMatrix =
-        matrix.reshaped(downsamplingFactor, (int)matrix.rows() * (int)matrix.cols() / downsamplingFactor);
+        Eigen::Map<Eigen::MatrixXf>(reshapedMatrixMap.data(), reshapedMatrixMap.rows(), reshapedMatrixMap.cols());
 
     std::function<float(float)> nan_to_zero_functor = nanToZero;
 
     reshapedMatrix = reshapedMatrix.unaryExpr(nan_to_zero_functor);
```

Comment:
Can we use a new variable here, instead of assigning to the old one?

Reply:
Yes. It is really difficult to come up with names for variables that are just part of the process. Is there any convention on this? I am inventing names to explain the operation.

Reply:
flattenedMatrixNansRemoved is the best I could think of.

Reply:
done